codegen/generator/ts/templates/functions.go

```diff
@@ -0,0 +1,159 @@
+package templates
+
+import (
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/iancoleman/strcase"
+
+	"github.com/dagger/dagger/codegen/introspection"
+)
+
+var (
+	FuncMap = template.FuncMap{
+		"CamelCase":              CamelCase,
+		"CommentToLines":         CommentToLines,
+		"FormatInputType":        FormatInputType,
+		"FormatOutputType":       FormatOutputType,
+		"FormatName":             FormatName,
+		"FieldOptionsStructName": FieldOptionsStructName,
+		"FieldFunction":          FieldFunction,
+		"PascalCase":             PascalCase,
+		"Solve":                  Solve,
+		"Subtract":               Subtract,
+	}
+)
+
+func PascalCase(name string) string {
+	return strcase.ToCamel(name)
+}
+
+func CamelCase(name string) string {
+	return strcase.ToLowerCamel(name)
+}
+
+func Solve(field introspection.Field) bool {
+	return field.TypeRef.IsScalar() || field.TypeRef.IsList()
+}
+
+// Subtract subtract integer a with integer b.
+func Subtract(a, b int) int {
+	return a - b
+}
+
+func CommentToLines(s string) []string {
+	split := strings.Split(s, "\n")
+	return split
+}
+
+// formatType formats a GraphQL type into Go
+// Example: `String` -> `string`
+func FormatInputType(r *introspection.TypeRef) string {
+	return formatType(r)
+}
+
+func FormatOutputType(r *introspection.TypeRef) string {
+	return formatType(r)
+}
+
+// formatType formats a GraphQL type into TypeScript
+// Example: `String` -> `string`
+func formatType(r *introspection.TypeRef) string {
+	var representation string
+	for ref := r; ref != nil; ref = ref.OfType {
+		switch ref.Kind {
+		case introspection.TypeKindList:
+			representation += "[]"
+		case introspection.TypeKindScalar:
+			switch introspection.Scalar(ref.Name) {
+			case introspection.ScalarString:
+				representation += "string"
+				return representation
+			case introspection.ScalarInt, introspection.ScalarFloat:
+				representation += "number"
+				return representation
+			case introspection.ScalarBoolean:
+				representation += "bool"
+				return representation
+			default:
+				// Custom scalar
+				representation += ref.Name
+				return representation
+			}
+		case introspection.TypeKindObject:
+			representation += FormatName(ref.Name)
+			return representation
+		case introspection.TypeKindInputObject:
+			representation += FormatName(ref.Name)
+			return representation
+		}
+	}
+
+	panic(r)
+}
+
+// FormatName formats a GraphQL name (e.g. object, field, arg) into a TS equivalent
+// Example: `fooId` -> `FooId`
+func FormatName(s string) string {
+	return s
```

Comment:
Right now nothing happens here, is it because it is still on draft?

Reply:
Yes, and I wanted to keep coherency with the Go SDK template functions.
Will see how it evolves. But for typescript, it doesn't need the transformation as the type comes from GraphQL and are already in PascalCase.